x-pack/plugin/transform/qa/single-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/transform/integration/TransformPivotRestIT.java

```diff
@@ -1006,6 +1008,86 @@ public void testPreviewTransformWithPipelineScript() throws Exception {
         );
     }
 
+    /**
+     * This test case makes sure that deprecation warnings from _search API are propagated to _preview API.
+     */
+    public void testPreviewTransformWithScriptedMetricUsingDeprecatedSyntax() throws Exception {
+        testTransformUsingScriptsUsingDeprecatedSyntax("POST", getTransformEndpoint() + "_preview");
+    }
+
+    /**
+     * This test case makes sure that deprecation warnings from _search API are propagated to PUT API.
+     */
+    public void testCreateTransformWithScriptedMetricUsingDeprecatedSyntax() throws Exception {
+        testTransformUsingScriptsUsingDeprecatedSyntax("PUT", getTransformEndpoint() + "script_deprecated_syntax");
+    }
+
+    private void testTransformUsingScriptsUsingDeprecatedSyntax(String method, String endpoint) throws Exception {
+        String transformIndex = "script_deprecated_syntax";
+        String config = "{"
+            + "  \"source\": {"
+            + "    \"index\": \""
+            + REVIEWS_INDEX_NAME
+            + "\","
+            + "    \"runtime_mappings\": {"
+            + "      \"timestamp-5m\": {"
+            + "        \"type\": \"date\","
+            + "        \"script\": {"
+            // We don't use "era" for anything in this script. This is solely to generate the deprecation warning.
+            + "          \"source\": \"def era = doc['timestamp'].value.era; emit(doc['timestamp'].value.millis)\""
+            + "        }"
+            + "      }"
+            + "    }"
+            + "  },"
+            + "  \"dest\": {"
+            + "    \"index\": \""
+            + transformIndex
+            + "\""
+            + "  },"
+            + "  \"pivot\": {"
+            + "    \"group_by\": {"
+            + "      \"timestamp\": {"
+            + "        \"date_histogram\": {"
+            + "          \"field\": \"timestamp-5m\","
+            + "          \"calendar_interval\": \"1m\""
+            + "        }"
+            + "      }"
+            + "    },"
+            + "    \"aggregations\": {"
+            + "      \"bytes.avg\": {"
+            + "        \"avg\": {"
+            + "          \"field\": \"bytes\""
+            + "        }"
+            + "      },"
+            + "      \"millis\": {"
+            + "        \"scripted_metric\": {"
+            + "          \"init_script\": \"state.m = 0\","
+            + "          \"map_script\": \"state.m = doc['timestamp'].value.millis;\","
+            + "          \"combine_script\": \"return state.m;\","
+            + "          \"reduce_script\": \"def last = 0; for (s in states) {last = s;} return last;\""
+            + "        }"
+            + "      }"
+            + "    }"
+            + "  }"
+            + "}";
+
+        final Request request = new Request(method, endpoint);
+        request.setJsonEntity(config);
+        try {
+            client().performRequest(request);
+        } catch (WarningFailureException e) {
```

Comment:
The test will pass if no warnings are generated.

Could it use `expectThrows` here?

```
    WarningFailureException e = expectThrows(WarningFailureException.class, () -> client().performRequest(request));
```

Then instead of a `catch` block the code currently within the `catch` block can just follow on after that `expectThrows`.  Then the test will fail if there are no warnings.

Reply:
Good point!
Done.